pages/contact/forms.py

```diff
@@ -0,0 +1,181 @@
+		    be selected.
+		website: Honeypot field, must remain empty.
+		contact_ts: Signed timestamp token to check submission timing.
+		contact_dsc: Token that must match the cookie for double-submit check.
+
+	Anti-spam strategy:
+		- Honeypot rejects naive bots filling hidden fields.
+		- TimestampSigner token rejects submissions that are too fast (<2s) or
+		  too old (>60 minutes).
+		- Double-submit cookie reduces scripted posts and replays.
+	"""
+
+	name = forms.CharField(min_length=2, max_length=100, required=False)
+	email = forms.EmailField(required=False)
+	message = forms.CharField(min_length=20, max_length=5000, widget=forms.Textarea)
+	suggestion = forms.BooleanField(required=False)
+	dm_support = forms.BooleanField(
+		required=False, label="Request for data management/sharing support"
+	)
+	other = forms.BooleanField(required=False)
+
+	# Anti-spam fields
+	website = forms.CharField(required=False, widget=forms.HiddenInput)
+	contact_ts = forms.CharField(widget=forms.HiddenInput, strip=False)
+	contact_dsc = forms.CharField(widget=forms.HiddenInput, strip=False)
+
+	# Internal state for logging (not exposed to users)
+	_blocked_reason: Optional[str] = None
+
+	def __init__(self, *args: Any, request=None, **kwargs: Any) -> None:
+		"""Initialise the form.
+
+		Args:
+			*args: Positional form args.
+			request: Optional HttpRequest to access cookies for token checks.
+			**kwargs: Keyword form args.
+		"""
+		super().__init__(*args, **kwargs)
+		self.request = request
+
+	def clean_email(self) -> str:
+		"""Validate email and prevent header injection.
+
+		Returns:
+			The sanitised email value.
+
+		Raises:
+			ValidationError: If CR/LF characters are present.
+		"""
+		value = self.cleaned_data.get("email", "")
+		if "\r" in value or "\n" in value:
+			raise ValidationError("Enter a valid email address.")
+		return value
+
+	def clean_message(self) -> str:
+		"""Validate message content against HTML and URL constraints.
+
+		Returns:
+			The validated message content.
+
+		Raises:
+			ValidationError: If HTML-like tags are present or URL count exceeds
+			    the allowed threshold.
+		"""
+		value = self.cleaned_data.get("message", "")
+		# Reject HTML-like content
+		if "<" in value or ">" in value:
+			raise ValidationError("Please remove HTML tags.")
+		# Cap URLs
+		url_count = len(URL_REGEX.findall(value))
+		if url_count > MAX_URLS_ALLOWED:
+			raise ValidationError("Too many links in the message.")
+		return value
+
+	def clean(self) -> dict[str, Any]:
+		"""Form-level validation for category, timing, and token checks.
+
+		Returns:
+			The cleaned data dictionary.
+
+		Raises:
+			ValidationError: On any anti-spam or category selection failure.
+		"""
+		cleaned = super().clean()
+
+		# At least one category must be selected
+		if not (
+			cleaned.get("suggestion")
+			or cleaned.get("dm_support")
+			or cleaned.get("other")
+		):
+			raise ValidationError("Please select at least one option.")
```

Comment:
While testing `MAX_TOKEN_AGE`, noticed this is getting triggered even if a category is selected.